### ISSUE TITLE
Microsoft.EntityFrameworkCore.Cosmos 3.0.0-preview4.19216.3 Value cannot be null.Parameter name: stream

### ISSUE THREAD

**Issue by dongshengfengniaowu** [dotnet/efcore#15473]
Query like this:
``` csharp
XXModel originEntity = await db.Set<XXModel o>().Where(x => x.ModelId == Id).FirstOrDefaultAsync();
```

SomeTimes Will Occur this:
```
     System.ArgumentNullException: Value cannot be null.
        Parameter name: stream
           at System.IO.StreamReader..ctor(Stream stream, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean leaveOpen)
           at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.DocumentAsyncEnumerable.AsyncEnumerator.MoveNext(CancellationToken cancellationToken)
           at Microsoft.EntityFrameworkCore.Cosmos.Query.Expressions.Internal.QueryShaperExpression.AsyncShaperEnumerable`1.AsyncShaperEnumerator.MoveNext(CancellationToken cancellationToken)
           at System.Linq.AsyncEnumerable.FirstOrDefault_[TSource](IAsyncEnumerable`1 source, CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\First.cs:line 144
           at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.TaskResultAsyncEnumerable`1.Enumerator.MoveNext(CancellationToken cancellationToken)
           at System.Linq.AsyncEnumerable.SelectEnumerableAsyncIterator`2.MoveNextCore(CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\Select.cs:line 106
           at System.Linq.AsyncEnumerable.AsyncIterator`1.MoveNext(CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\AsyncIterator.cs:line 98
           at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext(CancellationToken cancellationToken)","Exception":"Type: System.ArgumentNullException
Message: Value cannot be null.
Parameter name: stream
DataJson: {}
Stacktrace:
   at System.IO.StreamReader..ctor(Stream stream, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean leaveOpen)
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.DocumentAsyncEnumerable.AsyncEnumerator.MoveNext(CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Cosmos.Query.Expressions.Internal.QueryShaperExpression.AsyncShaperEnumerable`1.AsyncShaperEnumerator.MoveNext(CancellationToken cancellationToken)
   at System.Linq.AsyncEnumerable.FirstOrDefault_[TSource](IAsyncEnumerable`1 source, CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\First.cs:line 144
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.TaskResultAsyncEnumerable`1.Enumerator.MoveNext(CancellationToken cancellationToken)
   at System.Linq.AsyncEnumerable.SelectEnumerableAsyncIterator`2.MoveNextCore(CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\Select.cs:line 106
   at System.Linq.AsyncEnumerable.AsyncIterator`1.MoveNext(CancellationToken cancellationToken) in D:\a\1\s\Ix.NET\Source\System.Interactive.Async\AsyncIterator.cs:line 98
   at Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext(CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteSingletonAsyncQuery[TResult](QueryContext queryContext, Func`2 compiledQuery, IDiagnosticsLogger`1 logger, Type contextType)
   at 
```

**Comment by dongshengfengniaowu** [comment]
Any Response ?

**Comment by ajcvickers** [comment]
@dongshengfengniaowu This hasn't been triaged yet; that will happen next week.

**Comment by divega** [comment]
Note from triage: @dongshengfengniaowu we actually need more details in order to diagnose this. Can you provide a repro project?

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by kierenj** [comment]
We just reproduced this here - this exact error occurred when using the local Cosmos emulator, when the database was created (via the emulator), but no collection was in place.

Calling `_context.Database.EnsureExists()` (or was it `EnsureCreated`? I'm sure you know the one I mean) - then created the collection, and everything worked.

Using Fiddler, we were able to see that the server responded with a "not found" on the collection, but EF Core just threw the above error.

Hopefully that's helpful!
